app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/registration/SubscriptionRecoveryController.scala

```diff
@@ -214,7 +210,7 @@ class SubscriptionRecoveryController @Inject() (
       // Update Recovered Subscription Information
       _ <- updateSubscription(subscriptionInformation)
       // Update Email
-      //      _ <- updateEmail(journey, subscriptionInformation)  // TODO - ECC-307
+      _ <- updateEmail(subscriptionInformation)  // TODO - ECC-307
```

Comment:
So updating email is done only during Subscription recovery journey in this case?

If I interpret Yasar's comment correctly

> If the user comes to the Subscription journey to get access to ATAR and if we don't send the verified email to customs-data-store, and later on if they come again to get access to CDS, this time they will hit the short journey and we won't send the email to customs-data-store again

We'd need to update email somewhere earlier in the subscription journey?

Reply:
The scenario yasar mentions only applies if we did this just for cds services... It is currently done for everyone irrespective. Cds is treated just like any other service, which is consistent with what was recently agreed with the architects. Any service requesting an enrolment key (subscription) where the user already has a Eori and a CDS Contract in ETMP will go through the SubscriptionRecovery journey which is called once we realise  that a CDS contract already exist

Reply:
We verify the email early in the journey and cache it locally already (so no change)... We write to customs data sore and ETMP at the end when we confirm our details call reg 06, release a contract exist that may need updating, and so enter the recovery journey which completes things.